lib/addons/ebs-csi-driver/index.ts

```diff
@@ -15,44 +17,121 @@ const versionMap: Map<KubernetesVersion, string> = new Map([
 export class EbsCsiDriverAddOn extends CoreAddOn {
+  readonly ebsProps: EbsCsiDriverAddOnProps;
 
-    constructor(readonly options?: EbsCsiDriverAddOnProps) {
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName,
-            versionMap: defaultProps.versionMap,
-        });
-    }
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+      addOnName: defaultProps.addOnName,
+      version: options?.version ?? defaultProps.version,
+      saName: defaultProps.saName,
+    });
+
+    this.ebsProps = {
+      ...defaultProps,
+      ...options,
+    };
+  }
+
+  providePolicyDocument(clusterInfo: ClusterInfo): PolicyDocument {
+    return getEbsDriverPolicyDocument(
+      clusterInfo.cluster.stack.partition,
+      this.options?.kmsKeys
+    );
+  }
+
+  async deploy(clusterInfo: ClusterInfo): Promise<Construct> {
+    const baseDeployment = await super.deploy(clusterInfo);
+
+    const cluster = clusterInfo.cluster;
+    let updateSc: KubernetesManifest;
 
-    providePolicyDocument(clusterInfo: ClusterInfo) : PolicyDocument {
-        return getEbsDriverPolicyDocument(clusterInfo.cluster.stack.partition, this.options?.kmsKeys);
+    if (this.ebsProps.storageClass) {
+      // patch resource on cluster
+      const patchSc = new KubernetesPatch(
+        cluster.stack,
+        `${cluster}-RemoveGP2SC`,
+        {
+          cluster: cluster,
+          resourceName: "storageclass/gp2",
+          applyPatch: {
+            metadata: {
+              annotations: {
+                "storageclass.kubernetes.io/is-default-class": "false",
+              },
+            },
+          },
+          restorePatch: {
+            metadata: {
+              annotations: {
+                "storageclass.kubernetes.io/is-default-class": "true",
+              },
+            },
+          },
+        }
+      );
+
+      // Create and set gp3 StorageClass as cluster-wide default
+      updateSc = new KubernetesManifest(
+        cluster.stack,
+        `${cluster}-SetDefaultSC`,
+        {
+          cluster: cluster,
+          manifest: [
+            {
+              apiVersion: "storage.k8s.io/v1",
+              kind: "StorageClass",
+              metadata: {
+                name: "gp3",
+                annotations: {
+                  "storageclass.kubernetes.io/is-default-class": "true",
+                },
+              },
+              provisioner: "ebs.csi.aws.com",
+              reclaimPolicy: "Delete",
+              volumeBindingMode: "WaitForFirstConsumer",
+              parameters: {
+                type: "gp3",
+                fsType: "ext4",
+                encrypted: "true",
+              },
+            },
+          ],
+        }
+      );
+
+      patchSc.node.addDependency(baseDeployment);
+      updateSc.node.addDependency(patchSc);
+
+      return updateSc;
+    } else 

```

Comment:
minor: else clause is redundant, since you are returning updateSc.

Reply:
weird, it thinks that base deployment can be undefined. nm, as i said it is minor.